app/lib/mail/personalisation.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# This module is meant to be registered as an email interceptor on Action Mailer for developers
+# debugging and testing purposes.
+#
+# It adds the :subject_tags personalisation field:
+#    For non-production or test emails it adds a personalisation entry like this:
+#        subject_tags: "[local to:aaa@d1.md1,bbb@d2.md2]"
+#
+#    For production or test environments, the entry added is empty:
+#        subject_tags: ""
+#
+# Also, for non production or test environments, emails with a :subject personalisation field are modified to
+# preprend the app_name (from ENV['VCAP_APPLICATION'] || 'local') and the list of destination emails addresses
+# like this:
+#       subject: "[app_name to:aaa@d1.md1,bbb@d2.md2] original_subject_content"
+#
+module Mail
+  module Personalisation
+    class << self
+      def delivering_email(mail)
+        set_personalisation(mail, :subject_tags, subject_tags(mail))
+
+        # TODO: This enforces backward compatibility with existing emails and should be deleted as soon as all the emails
+        # are modified to use `subject_tags` instead
+        override_personalisation(mail, :subject) { |subject| new_subject(mail, subject) } if enabled?
+      end
+
+    private
+
+      def app_name
+        ENV["VCAP_APPLICATION"] ? JSON.parse(ENV["VCAP_APPLICATION"])["application_name"] : "local"
```

Comment:
could we drop the `ENV["VCAP_APPLICATION"]` as this was for only PaaS specific implementation and doesn't apply in the new world?

Reply:
Sure. Is there a way to get the name of the app in the new world?

Reply:
`Rails.env` probably makes most sense.

Reply:
Yes, that's was my alternative if we don't have anything better.

Reply:
yea I agree, if we want to go with this, I think we're missing the target email logic from the new class, if we want to drop it but keep the personalised tags even better